forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/TppRegistrationService.java

```diff
@@ -251,6 +258,27 @@ public void verifyTPPRegistrationRequestAgainstSSA(RegistrationRequest registrat
 
     public Tpp registerTpp(ApiClientIdentity clientIdentity, RegistrationRequest oidcRegistrationRequest)
             throws DynamicClientRegistrationException {
+        
+        Optional<DirectorySoftwareStatement> jti = tppStoreService.findByOrganisationId(clientIdentity.getUsername())
+            .stream()
+            .map(Tpp::getSsa)
+            .map(ssa -> {
+                try {
+                    return objectMapper.readValue(ssa, DirectorySoftwareStatement.class);
```

Comment:
Let's change the existing TPP records to store the DirectorySoftwareStatement object rather than the serialised JSON string. That will make this easier.